src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/utilities/HeapHprofBinWriter.java

```diff
@@ -1354,7 +1364,7 @@ public SegmentedOutputStream(OutputStream out) {
          */
         @Override
         public synchronized void write(int b) throws IOException {
-           if (segmentMode) {
+           if (segmentMode && !unbufferedMode) {
```

Comment:
It is not clear why the condition` !unbufferedMode` is additionally checked here and several places below. The `SegmentedOutputStream` constructor always sets `false` to to `unbufferedMode` field. Is it possible that `segmentMode` can be also unbuffered?